crates/rome_js_formatter/src/utils/mod.rs

```diff
@@ -386,23 +386,270 @@ pub(crate) fn format_with_semicolon(
+                            // The second backslash is at the end.
+                            && start + 2 <= bytes.len()
+                        {
+                            match signal {
+                                CharSignal::Keep => {
+                                    reduced_string.push('\\');
+                                }
+                                _ => {
+                                    reduced_string.push(alternate_quote.as_char());
+                                    signal = CharSignal::AlreadyPrinted;
+                                }
+                            }
+                        } else {
+                            // The next character is another backslash, let's signal
+                            // the next iteration that it should keep it in the final string
+                            if bytes[start + 1] == b'\\' {
+                                signal = CharSignal::Keep;
+                            }
+                            // fallback, keep the backslash
+                            reduced_string.push('\\');
+                        }
+                    } else {
+                        // fallback, keep the backslash
+                        reduced_string.push('\\');
+                    }
+                }
+                _ => unreachable!("We checked already the presence of a backslash"),
+            }
+        }
+    }
+
+    reduced_string.push_str(&raw_content[last_end..raw_content.len()]);
+    reduced_string
+}
+
+const POSSIBLE_QUOTES: [char; 2] = ['\'', '"'];
+
+#[derive(Eq, PartialEq)]
+pub(crate) enum LiteralType {
+    Directive,
+    String,
+}
+
 pub(crate) fn format_string_literal_token(
     token: JsSyntaxToken,
     formatter: &Formatter<JsFormatOptions>,
+    literal_type: LiteralType,
 ) -> FormatElement {
-    let quoted = token.text_trimmed();
-    let (primary_quote_char, secondary_quote_char) = match formatter.options().quote_style {
-        QuoteStyle::Double => ('"', '\''),
-        QuoteStyle::Single => ('\'', '"'),
+    let literal = token.text_trimmed();
+
+    let content = match literal_type {
+        LiteralType::Directive => {
+            let raw_content = &literal[1..literal.len() - 1];
+
+            if raw_content.contains(['"', '\'']) {
+                normalize_newlines(literal, ['\r'])
+            } else {
+                let quote_style = formatter.options().quote_style;
+                let final_content = format!(
+                    "{}{}{}",
+                    quote_style.as_char(),
+                    raw_content,
+                    quote_style.as_char()
+                );
+
+                Cow::Owned(normalize_newlines(&final_content, ['\r']).into_owned())
+            }
+        }
+
+        LiteralType::String => {
+            // literal, in our syntax, might not have quotes so we need to check if they start with a possible quote
+            // is so, they are eligible for possible string manipulation
```

Comment:
```suggestion
            // literal, in our syntax, might not have quotes so we need to check if they start with a possible quote
            // and if so, they are eligible for possible string manipulation
```

When is it possible for a string literal to not start with quotes? Would you mind making an example here.

Reply:
An example:

```
export { a as "b" };
export { a as b };
```